### DISCUSSION TITLE
record on demand using API

### DISCUSSION THREAD

**Discussion by skaface** [bluenviron/mediamtx#2922] [Questions]
### Question

I am using mediamtx to provide streams from security cameras, and currently the videos are recorded on demand from another computer receiving the streams. Is it possible to send a command to the mediamtx API with curl to tell it to start/stop recording of a stream?

**Comment by aler9** [comment]
Hello, recording a specific stream can be toggled through the API with

```
curl -X PATCH -d '{"record": true}' http://localhost:9997/v3/config/paths/patch/my_path_name
```

**Reply by KorsAir1987** [reply]
This only seems to work if the stream's path is hardcoded in the configuration file, not for streams pushed dynamically to the media server.

**Reply by aler9** [reply]
@KorsAir1987 you have to create a dedicated path before the stream is pushed to the server or alternatively wait for https://github.com/bluenviron/mediamtx/issues/3718

**Reply by dolldreamdaker-hub** [reply]
> @KorsAir1987您必须在将流推送到服务器之前创建专用路径，或者等待#3718

如果动态路径支持直接api录制就好了
